### PR TITLE
[connectors] - feature(microsoft): implement PDF file handling

### LINKED ISSUE #5744: [MS connector] PDF extraction
Extract and chunk pdf content

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at adding support for PDF file in Microsoft connector.

**References:**
- https://github.com/dust-tt/dust/issues/5744

### CODE REVIEW

**Comment by JulesBelveze** [comment]
On hold until we introduce Apache Tika (has pdf2text) is a bit inconsistent and flaky

File: connectors/package-lock.json [hunk 1]
```diff
@@ -90,6 +90,7 @@
         "eslint-plugin-simple-import-sort": "^12.1.0",
         "eventsource-parser": "^1.1.1",
         "hot-shots": "^10.0.0",
+        "htmlparser2": "^9.1.0",
```
Comment (flvndvd) [inline connectors/package-lock.json:93]: Can be removed.

File: connectors/src/connectors/microsoft/temporal/activities.ts [hunk 2]
```diff
@@ -748,3 +760,52 @@ async function handleCsvFile(
   }
   return true;
 }
+
+async function handleTextExtraction(
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/temporal/activities.ts:764]: This is so similar to Google Drive that I'll push to make this a common function!

**Review by flvndvd — APPROVED** [review]
LGTM, make sure to fix your dependencies. Left one comment as I feel we duplicate a lot of code.

File: connectors/src/connectors/microsoft/temporal/activities.ts (hunk 2 above)
Comment (JulesBelveze) [reply connectors/src/connectors/microsoft/temporal/activities.ts:764]: Added a ticket here to refacto and share functions across GDrive and Microsoft https://github.com/dust-tt/dust/issues/6283

File: connectors/package-lock.json (hunk 1 above)
Comment (JulesBelveze) [reply connectors/package-lock.json:93]: @flvndvd `types` does depend on `htmlparser2` right? Or did we remove the dependency?

Comment (flvndvd) [reply connectors/package-lock.json:93]: Yes `types` does, but `connectors` should not need it.

Comment (JulesBelveze) [reply connectors/package-lock.json:93]: As per IRL: this is in `../types` deps

**Pull request merged**